designs/Consolidate-NuGetConfig_OnLinux.md

```diff
@@ -0,0 +1,216 @@
+
+# Spec Name
+
+* Status: In Review
+* Author(s): [Heng Liu](https://github.com/heng-liu)
+* Issue: [4413](https://github.com/NuGet/Home/issues/4413) [MacOS, Linux] NuGet writes the NuGet.Config in ~/.config folder, but dotnet restore reads from ~/.nuget 
+
+## Problem Background
+
+On Linux/Mac, NuGet has inconsistent paths of user wide NuGet.Config file for different NuGet tools. Dotnet CLI uses hard coded `~/.nuget` folder, so the user wide NuGet.Config path is `~/.nuget/NuGet/NuGet.Config`). While Mono uses the folder name from environment variable `APPDATA`, so by default, the user wide NuGet.Config file path is `~/.config/NuGet/NuGet.Config`). This is confusing and we should consolidate the two into one location. 
+
+After discussing in NuGet team and checking with Visual Studio Mac/Mono team, we believe we should change the dotnet CLI path to match that of Mono’s, for the following reason:
+ * On Linux/Mac, users usually get application data from environment variable XDG_DATA_HOME, which has the default value of ~/.config. Getting application data from ~/.nuget will cause inconsistency user experience between NuGet and other applications.
+
+For the sake of simplicity, we will refer to `~/.nuget/NuGet/NuGet.Config` as legacy path, and refer to `~/.config/NuGet/NuGet.Config` as consolidated path in the following.
+
+## Who are the customers
+Dotnet CLI users who work on Linux and Mac.
+
+## Goals
+Make dotnet CLI user-wide config file path consistent with Mono on Linux/Mac and break as less dotnet CLI users as possible.
+
+## Non-Goals
+Completely get rid of the legacy path.
+
+## Background knowledge
+On Linux/Mac, user wide NuGet.Config file is created when trying to get the user wide NuGet.Config file at the beginning phase of running `dotnet restore` command(and many other dotnet commands with implicit restore if not specifying `--no-restore` option, like `dotnet new`, `dotnet build`, `dotnet run`, `dotnet test`, `dotnet publish`, and `dotnet pack`), or `dotnet msbuild -t:restore` command, but the user wide NuGet.Config file does not exist. Then the user wide NuGet.Config will be created with package source of nuget.org, as the only default package source. 
+
+## Potential Solutions overview 
+### Solution 1: Use only consolidated path
```

Comment:
There's another option, very similar to this Solution 1:  Output a warning to the customer when the legacy path exists, and let them take manual action, if they want to. We should include a URL to a docs page in the warning, and that docs page can suggest manually merging the two nuget.config files into one (that was we don't have to write complex code that will try to do it automatically), and suggest setting up a symbolic link if they use global.json.

This has the least effort on our part, allowing us to implement is the most quickly.